### PR TITLE
CRM-21166: WP-CLI commands fail due to whitespace in regex  Remove wh…

### PR DESCRIPTION
**Pull request by kcristiano**
…itespace from regex on parseDSN function.

### CODE REVIEW

**Comment by kcristiano** [comment]
This PR completes the cleanup in https://github.com/civicrm/civicrm-wordpress/pull/114  @totten is it possible to get this in 4.7.25 rc?

**Comment by totten** [comment]
Yes, I think fixing a recent regression in the RC makes sense. Edited the PR to target the `4.7.25-rc`. Confirmed that this patch removes some `preg_match` errors when running `wp cv sql-dump`. If there's no other feedback in the next day, then let's merge into the RC.

(Note to self: after merging, we'll need to merge-forward to `master`.)

CC @jonesinator @agh1 because the JIRA description references related code from #110 and #114.

**Comment by agh1** [comment]
Ugh sorry--thanks for catching this!  I suspect that it got introduced by the editor and/or PHPCS as it tries to enforce WP style with spaces inside of parentheses.

File: wp-cli/civicrm.php [hunk 1]
```diff
@@ -1163,7 +1163,7 @@ private static function parseDSN( $dsn ) {
 
 			// Find protocol and hostspec
 
-			if ( preg_match( '|^( [^( ]+ )\( ( .*? )\ )/?( .*? )$|', $dsn, $match ) ) {
+			if ( preg_match( '|^([^(]+)\((.*? )\)/?(.*?)$|', $dsn, $match ) ) {
```
Comment (agh1) [inline wp-cli/civicrm.php]: Is there one extra space to remove (after the first question mark)?

Comment (kcristiano) [reply wp-cli/civicrm.php]: missed that.  Thanks.  Updating the PR

**Comment by eileenmcnaughton** [comment]
@agh1 @totten looks like this has gone cold - good to merge?

**Comment by agh1** [comment]
@eileenmcnaughton I think it is--I've applied these changes already in one spot where I encountered the bug.

**Pull request merged**